build.gradle

```diff
@@ -127,6 +127,16 @@ if (currentVersion.bugfix == 0) {
   }
 }
 
+// build metadata from previous build, contains eg hashes for bwc builds
+String buildMetadataValue = System.getenv('BUILD_METADATA')
+if (buildMetadataValue == null) {
+  buildMetadataValue = ''
+}
+Map<String,String> buildMetadataMap = buildMetadataValue.tokenize(';').collectEntries {
+  def (String key, String value) = it.split('=')
+  return [key: value]
```

Comment:
I think that you need to wrap `key` in parentheses here (`(key)`) otherwise this will be interpreted as a key literally named `key`. Alternatively, you can write this as:

`  return [key, value]`